src/libertem/io/dataset/dm_single.py

```diff
@@ -0,0 +1,370 @@
+import os
+import warnings
+import typing
+from typing import Dict, Any, Tuple, Optional
+import logging
+
+import ncempy.io.dm
+import numpy as np
+
+from libertem.common.math import prod
+from libertem.common import Shape
+from .base import BasePartition, DataSetException, DataSetMeta, File, IOBackend
+from .dm import DMDataSet, SingleDMDatasetParams, DMFileSet
+
+log = logging.getLogger(__name__)
+
+if typing.TYPE_CHECKING:
+    from numpy import typing as nt
+    from libertem.executor.base import JobExecutor
```

Comment:
```suggestion
    from libertem.common.executor import JobExecutor
```

(to try to keep things license-clean)